File: Daily/Rust/Leetcode1846-减小和重新排列数组之后的最大值.rs
impl Solution {
    pub fn maximum_element_after_decrementing_and_rearranging(mut arr: Vec<i32>) -> i32 {
        let n = arr.len();
        arr.sort_unstable();
        arr[0] = 1;
        for i in 1..n{
            arr[i] = arr[i].min(arr[i - 1] + 1);
        }
        arr[n - 1]
    }
}

File: Leetcode/PY/Leetcode1846-减小和重新排列数组之后的最大值.py
from typing import List

class Solution:
    def maximumElementAfterDecrementingAndRearranging(self, arr: List[int]) -> int:
        n = len(arr)
        arr.sort()
        arr[0] = 1
        for i in range(1,n):
            arr[i] = min(arr[i],arr[i - 1] + 1)
        return arr[-1]